native_client/ctcdecode/path_trie.h

```diff
@@ -9,6 +9,28 @@
 
 #include "fst/fstlib.h"
 #include "alphabet.h"
+#include "object_pool.h"
+
+/* Tree structure with parent and children information
+ * It is used to store the timesteps data for the PathTrie below
+ */
+template<class DataT>
+struct TreeNode {
+    TreeNode<DataT>* parent;
+    std::vector<std::unique_ptr< TreeNode<DataT>, godefv::memory::object_pool_deleter_t<TreeNode<DataT>> >> children;
+
+    DataT data;
+
+    TreeNode(TreeNode<DataT>* parent_, DataT const& data_): parent{parent_}, data{data_} {}
+};
+
+template<class NodeDataT, class ChildDataT>
+TreeNode<NodeDataT>* add_child(TreeNode<NodeDataT>* node, ChildDataT&& data_);
+
+template<class DataT>
+std::vector<DataT> get_history(TreeNode<DataT> const*, TreeNode<DataT> const* = nullptr);
```

Comment:
Please name the parameters and briefly document them, just a couple lines is OK.